src/main/java/com/conveyal/datatools/manager/models/FeedVersionDeployed.java

```diff
@@ -0,0 +1,239 @@
+    public LocalDate endDate;
+
+    public FeedVersionDeployed() {
+    }
+
+    public FeedVersionDeployed(Document feedVersionDocument) {
+        this.id = feedVersionDocument.getString("_id");
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMdd");
+        Document validationResult = (Document) feedVersionDocument.get("validationResult");
+        if (validationResult != null) {
+            String first = validationResult.getString("firstCalendarDate");
+            String last = validationResult.getString("lastCalendarDate");
+            this.startDate = (first == null) ? null : LocalDate.parse(first, formatter);
+            this.endDate = (last == null) ? null : LocalDate.parse(last, formatter);
+        }
+    }
+
+    /**
+     * Get the deployed feed version from the pinned deployment for this feed source.
+     */
+    public static FeedVersionDeployed getFeedVersionFromPinnedDeployment(String projectId, String feedSourceId) {
+        /*
+            Note: To test this script:
+                1) Comment out the call to tearDownDeployedFeedVersion() in FeedSourceControllerTest -> tearDown().
+                2) Run FeedSourceControllerTest to created required objects referenced here.
+                3) Once complete, delete documents via MongoDB.
+                4) Uncomment the call to tearDownDeployedFeedVersion() in FeedSourceControllerTest -> tearDown().
+                5) Re-run FeedSourceControllerTest to confirm deletion of objects.
+
+            db.getCollection('Project').aggregate([
+            {
+                $match: {
+                    _id: "project-with-pinned-deployment"
+                }
+            },
+            {
+                $lookup:{
+                    from:"Deployment",
+                    localField:"pinnedDeploymentId",
+                    foreignField:"_id",
+                    as:"deployment"
+                }
+            },
+            {
+                $lookup:{
+                    from:"FeedVersion",
+                    localField:"deployment.feedVersionIds",
+                    foreignField:"_id",
+                    as:"feedVersions"
+                }
+            },
+            {
+                $unwind: "$feedVersions"
+            },
+            {
+                "$replaceRoot": {
+                    "newRoot": "$feedVersions"
+                }
+            },
+            {
+                $match: {
+                    feedSourceId: "feed-source-with-pinned-deployment-feed-version"
+                }
+            },
+            {
+                $sort: {
+                    lastUpdated : -1
+                }
+            },
+            {
+               $limit: 1
+            }
+            ])
+         */
+        List<Bson> stages = Lists.newArrayList(
+            match(
+                in("_id", projectId)
+            ),
+            lookup("Deployment", "pinnedDeploymentId", "_id", "deployment"),
```

Comment:
Is there no way to check for the pinned deployment existing as part of this call?

Reply:
Short answer, probably. See my response to question above, but I would need to spend more time on this.

Reply:
Let's make it a separate PR!